core/src/col_metadata.rs

```diff
@@ -40,7 +43,8 @@ impl MongoColMetadata {
         field_schema: Schema,
         is_nullable: ColumnNullability,
     ) -> MongoColMetadata {
-        let bson_type_info: BsonTypeInfo = field_schema.into();
+        let bson_type_info: BsonTypeInfo = (&field_schema).into();
```

Comment:
changing these `From` impls to work on refs saves us a clone.

Reply:
smart!